File: app/src/main/java/me/pwcong/jpstart/component/fragment/BannerFragment.java
package me.pwcong.jpstart.component.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

import butterknife.BindView;
import me.pwcong.jpstart.R;
import me.pwcong.jpstart.conf.Constants;
import me.pwcong.jpstart.mvp.bean.BannerItem;

/**
 * Created by Pwcong on 2016/10/10.
 */

public class BannerFragment extends BaseFragment {

    @BindView(R.id.iv_banner)
    ImageView mImageView;

    private int banner;

    public static BannerFragment newInstance(BannerItem item){

        Bundle arguments = new Bundle();
        arguments.putInt(Constants.IMG_BANNER,item.getBanner());
        BannerFragment fragment=new BannerFragment();
        fragment.setArguments(arguments);

        return fragment;

    }


    @Override
    protected int getViewId() {
        return R.layout.fragment_banner;
    }

    @Override
    protected void initVariable(@Nullable Bundle savedInstanceState) {

        banner=getArguments().getInt(Constants.IMG_BANNER);

    }

    @Override
    protected void doAction() {
        Glide.with(getContext()).asBitmap().load(banner).into(mImageView);
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/view/fragment/BannerFragment.kt
package com.github.pwcong.jpstart.view.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.github.pwcong.jpstart.constants.Constants
import com.github.pwcong.jpstart.databinding.FragmentBannerBinding
import com.github.pwcong.jpstart.mvp.bean.BannerItem

class BannerFragment : BaseFragment<FragmentBannerBinding>() {

    private lateinit var mImageView: ImageView

    private var banner = 0

    override fun initViewBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): FragmentBannerBinding {
        return FragmentBannerBinding.inflate(inflater, container, false)
    }

    override fun init(savedInstanceState: Bundle?) {
        mImageView = viewBinding.ivBanner
        banner = requireArguments().getInt(Constants.IMG_BANNER)

        context?.let { Glide.with(it).asBitmap().load(banner).into(mImageView) }
    }

    companion object {
        fun newInstance(item: BannerItem): BannerFragment {
            val arguments = Bundle()
            arguments.putInt(Constants.IMG_BANNER, item.banner)
            val fragment = BannerFragment()
            fragment.arguments = arguments

            return fragment
        }
    }
}
